bool avaiable
float32 cx
float32 cy
float32 w